lib/client.js

```diff
@@ -603,53 +601,15 @@ Client.prototype.execute = function (query, params, options, callback) {
 };
 
 /**
- * Executes a graph query.
- * <p>It returns a <code>Promise</code> when a <code>callback</code> is not provided.</p>
- * @param {String} query The gremlin query.
- * @param {Object|null} [parameters] An associative array containing the key and values of the parameters.
- * @param {GraphQueryOptions|null} [options] The graph query options.
- * @param {Function} [callback] Function to execute when the response is retrieved, taking two arguments:
- * <code>err</code> and <code>result</code>. When not defined, the method will return a promise.
- * @example <caption>Promise-based API, using async/await</caption>
- * const result = await client.executeGraph('g.V()');
- * // Get the first item (vertex, edge, scalar value, ...)
- * const vertex = result.first();
- * console.log(vertex.label);
- * @example <caption>Callback-based API</caption>
- * client.executeGraph('g.V()', (err, result) => {
- *   const vertex = result.first();
- *   console.log(vertex.label);
- * });
- * @example <caption>Iterating through the results</caption>
- * const result = await client.executeGraph('g.E()');
- * for (let edge of result) {
- *   console.log(edge.label); // created
- * });
- * @example <caption>Using result.forEach()</caption>
- * const result = await client.executeGraph('g.V().hasLabel("person")');
- * result.forEach(function(vertex) {
- *   console.log(vertex.type); // vertex
- *   console.log(vertex.label); // person
- * });
- * @see {@link ExecutionProfile} to reuse a set of options across different query executions.
+ * @deprecated Not supported by the driver. Usage will throw an error.
  */
 Client.prototype.executeGraph = function (
     query,
     parameters,
     options,
     callback,
 ) {
-    callback = callback || (options ? options : parameters);
-
-    if (typeof callback === "function") {
-        parameters = typeof parameters !== "function" ? parameters : null;
-        return promiseUtils.toCallback(
-            this._graphExecutor.send(query, parameters, options),
-            callback,
-        );
-    }
-
-    return this._graphExecutor.send(query, parameters, options);
+    throwNotSupported("function", "Client.executeGraph");
```

Comment:
I suggest another layer of helper functions:
`throwNotSupportedFunction(functionName)` := `throwNotSupportedEntity("function", functionName)`
`throwNotSupportedClass(className)` := `throwNotSupportedEntity("class", className)`
This way we type less for each call, leaving less space for typos.

Reply:
The change should be easy to be done in all places by Search + Replace All.
- `throwNotSupported("function", ` -> `throwNotSupportedFunction(`

Reply:
This is no longer relevant.